### PR TITLE
Add session linking documentation

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Adds session linking documentation. Also updates some older gifs and images

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **Documentation**
	- Enhanced the user guide with new content on viewing samples and groups within a session, including how to open a sample by ID and manage group datasets.
	- Added practical examples and context to improve user understanding of session properties in Python.
- **New Features**
	- Introduced a new method to load a saved workspace by name within the session.
- **Improvements**
	- Reorganized parameters in the launch functionality for better clarity and usability.
	- Updated documentation to reflect changes in parameter organization and enhance user experience.

### CODE REVIEW

File: docs/source/user_guide/app.rst [hunk 1]
```diff
@@ -161,6 +161,63 @@ first 10 samples in the dataset sorted by their `uniqueness` field:
...
+.. code-block:: python
+    :linenos:
+
+    import fiftyone as fo
+    import fiftyone.zoo as foz
+
+    dataset = foz.load_zoo_dataset("quickstart-groups")
+    group_id = dataset.last().id
+
+    session = fo.launch_app(dataset, group_id=group_id)
```
Comment (brimoor) [inline docs/source/user_guide/app.rst:208]: @benjaminpkane FYI when I run this locally the group data doesn't successfully load in the modal (same when setting `session.group_id = XXX` below

Comment (benjaminpkane) [reply docs/source/user_guide/app.rst:208]: My mistake, it should be `dataset.last().group.id`. Will update

Comment (brimoor) [reply docs/source/user_guide/app.rst:208]: oops that was my bad. Shakespeare edit fail. I've corrected the docs and it works now 😅

**Review by brimoor — DISMISSED** [review]
LGTM 🚀

**Review by brimoor — APPROVED** [review]

**Pull request merged**